File: Data Structures & Algorithms/two-integer-sum/submission-0.js
class Solution {
  twoSum(nums, target) {
    const map = {};
    for (let i = 0; i < nums.length; i++) {
      map[nums[i]] = i;
    }
    
    for (let i = 0; i < nums.length; i++) {
      const complement = target - nums[i];
      if (complement in map && map[complement] !== i) {
        return [i, map[complement]];
      }
    }
    return [];
  }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
public int[] ProductExceptSelf(int[] nums) {
    int[] result = new int[nums.Length];
    int pointer = 0;

    while (pointer < nums.Length) {
        int curr = 1;
        for (int j = 0; j < nums.Length; j++) {
            if (j != pointer) {
                curr *= nums[j];
            }
        }
        result[pointer] = curr;
        pointer += 1;
    }

    return result;
}
}
